src/logics/ArithLogic.cc

```diff
@@ -560,8 +560,11 @@ PTRef ArithLogic::mkMinus(vec<PTRef> && args)
     if (args.size() == 1) {
         return mkNeg(args[0]);
     }
-    assert(args.size() == 2);
-    if (args.size() > 2) { throw OsmtApiException("Too many terms provided to LALogic::mkNumMinus"); }
+    if (args.size() > 2) { // Minus is left associative according to SMT-LIB
+        PTRef last = args.last();
+        args.pop();
+        return mkMinus(mkMinus(std::move(args)), last);
```

Comment:
We could do something like
```
for (int i = 1; i < args.size(); i++)
  args[i] = mkNeg(args[i]);
```
and pass `args` directly to `mkPlus`.  It doesn't look like it'd take much more space and we can get rid of special handling?  I guess there's a more elegant way than the `for` loop with an index, but anyway.

Reply:
Ah, yes, you are right! For some reason I thought that we cannot flatten such expressions, but, yes, we can!